src/awkward/forms/numpyform.py

```diff
@@ -11,14 +12,26 @@
 np = NumpyMetadata.instance()
 
 
-def from_dtype(dtype, parameters=None):
+def from_dtype(dtype, parameters=None, *, time_units_as_parameter: bool = UNSET):
     if dtype.subdtype is None:
         inner_shape = ()
     else:
         inner_shape = dtype.shape
         dtype = dtype.subdtype[0]
 
-    if issubclass(dtype.type, (np.datetime64, np.timedelta64)):
+    if time_units_as_parameter is UNSET:
+        time_units_as_parameter = True
+
+    if time_units_as_parameter:
+        deprecate(
+            "from_dtype conversion of datetime units to generic `datetime` types is deprecated, "
```

Comment:
```suggestion
            "from_dtype conversion of temporal units to generic `datetime64` and `timedelta64` types is deprecated, "
```

Just to be a little clearer.